internal/client/echo_client.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"github.com/afex/hystrix-go/hystrix"
+	"github.com/go-kit/kit/log"
+	"net/http"
+	"os"
+	"time"
+)
+
+const (
+	url           = "http://localhost:8080/test"
+	cmd           = "echo_cmd"
+	streamAddress = "localhost:8088"
+)
+
+func StartEchoClient() {
+	var logger log.Logger
+	logger = log.NewLogfmtLogger(os.Stdout)
+
+	hystrix.ConfigureCommand(cmd, hystrix.CommandConfig{
+		Timeout:               1000,
+		MaxConcurrentRequests: 100,
+		ErrorPercentThreshold: 30,
+	})
+
+	streamHandler := hystrix.NewStreamHandler()
+	streamHandler.Start()
+	go http.ListenAndServe(streamAddress, streamHandler)
+
+	var client EchoClient
+	client = NewEchoClient(url)
+	client = retryMiddleware(logger)(client)
+	client = circuitBreakerMiddleware(cmd, logger)(client)
```

Comment:
good point to use both circuit breaker & retry
but why do we choose retry inside circuit breaker instead of circuit breaker inside retry? If you read the grab engineering blog which is included in assignment, circuit breaker inside retry is more preferable.

Reply:
Thank you for the suggestion! I will read this later